client/dashboard/sites/settings/danger-zone.tsx

```diff
@@ -74,10 +76,14 @@ const SiteDeleteAction = ( { site }: { site: Site } ) => {
 	return (
 		<>
 			<ActionList.ActionItem
-				title={ __( 'Delete site' ) }
-				description={ __(
-					'Delete all your posts, pages, media, and data, and give up your site’s address.'
-				) }
+				title={ site.is_wpcom_staging_site ? __( 'Delete staging site' ) : __( 'Delete site' ) }
```

Comment:
In my opinion, it will be much clearer if we just branch the code based on whether or not the site is staging site. E.g.:

```tsx
const button = ...;

return site.is_wpcom_staging_site
  ? ( 
      <>
         <ActionList.ActionItem ..
         { isOpen && <StagingSiteDeleteModal ...
       </>
    ) 
   : (
      <>
         <ActionList.ActionItem ..
         { isOpen && <SiteDeleteModal ...
       </>
   )
);
```

There might be some duplication but the code is easier to follow. What do you think?